src/main/java/com/sequenceiq/cloudbreak/service/stack/handler/UpdateInstancesRequestHandler.java

```diff
@@ -136,21 +143,40 @@ private void downScaleStack(UpdateInstancesRequest request, Stack stack) throws
                     futures.add(submit);
                     if (provisionUtil.isRequestFull(stack, futures.size() + 1)) {
                         Map<FutureResult, List<ResourceRequestResult>> result = provisionUtil.waitForRequestToFinish(stack.getId(), futures);
-                        provisionUtil.checkErrorOccurred(result);
+                        failedResourceList.addAll(result.get(FutureResult.FAILED));
                         futures = new ArrayList<>();
                     }
                 }
                 Map<FutureResult, List<ResourceRequestResult>> result = provisionUtil.waitForRequestToFinish(stack.getId(), futures);
-                provisionUtil.checkErrorOccurred(result);
+                failedResourceList.addAll(result.get(FutureResult.FAILED));
             }
+            instanceIds = checkErrorOccurred(failedResourceList, instanceIds);
             if (!stackRepository.findById(stack.getId()).isStackInDeletionPhase()) {
                 stackUpdater.removeStackResources(stack.getId(), deleteContextObject.getDecommissionResources());
                 LOGGER.info("Terminated instances in stack: '{}'", instanceIds);
                 LOGGER.info("Publishing {} event.", ReactorConfig.REMOVE_INSTANCES_COMPLETE_EVENT);
-                reactor.notify(ReactorConfig.REMOVE_INSTANCES_COMPLETE_EVENT, Event.wrap(new StackUpdateSuccess(stack.getId(), true,
-                        instanceIds, request.getInstanceGroup())));
+                reactor.notify(ReactorConfig.REMOVE_INSTANCES_COMPLETE_EVENT,
+                        Event.wrap(new StackUpdateSuccess(stack.getId(), true, instanceIds, request.getInstanceGroup())));
+            }
+        }
+    }
+
+    private Set<String> checkErrorOccurred(List<ResourceRequestResult> failedResourceList, Set<String> instanceIds) {
```

Comment:
The name is misleading, should be something like: filterFailedResources. Otherwise it is too complicated with unnecessary iterations, can be simplified :
```java
Set<String> result = new HashSet<>(instanceIds);
        for (ResourceRequestResult requestResult : failedResourceList) {
            for (Resource resource : requestResult.getResources()) {
                String resourceName = resource.getResourceName();
                if (result.contains(resourceName)) {
                    result.remove(resourceName);
                }
            }
        }
        return result;
```